pkg/resolve/resolve.go

```diff
@@ -136,6 +149,70 @@ type Opts struct {
 	ProviderToken string
 }
 
+// getRemoteTasks will get remote tasks from annotations on pipelinerun, remote
+// pipeline and remote tasks from remote pipeline unless already requested in
+// PipelineRun
+// This means the PipelineRun task will always take precedence and this allows
+// for example overriding the tasks from the global template with a task locally.
+func getRemoteTasks(ctx context.Context, rt *matcher.RemoteTasks, types *TektonTypes) error {
+	for _, pipelinerun := range types.PipelineRuns {
+		if len(pipelinerun.GetObjectMeta().GetAnnotations()) == 0 {
+			continue
+		}
+
+		remoteTasks, err := rt.GetTaskFromAnnotations(ctx, pipelinerun.GetObjectMeta().GetAnnotations())
+		if err != nil {
+			return fmt.Errorf("error getting remote task from pipelinerun annotations: %w", err)
+		}
+
+		// Merge remote tasks with local tasks
+		for _, remoteTask := range remoteTasks {
+			if alreadySeen(types.Tasks, remoteTask) {
```

Comment:
Should not this will give preference to local tasks, then remote task?

Reply:
This is what happen here, the precedence works like this:

The tasks in the tekton directory (we get all files in .tekton dir and subdir and parses it)
The tasks in the PipelineRun annotations
The tasks in the Pipeline annotations

Maybe we should make the tasks in PipelineRun annotation precedence over tekton directory tasks

Reply:
from the doc i understood that, the one on annotation in given preference

Reply:
I refactored the PR to make it more idiomatic, moved it to its own files and improved the docs to make it clear...  Please let me know what do you think